Loader: Intro, `fabric.mod.json`, Dependency Overrides (#557)
FabricMC/fabric-docs: players/troubleshooting/dependency-overrides.md

Reviewer: I haven't fully reviewed this file yet, but I don't want to impede this PR further. I think we can merge this, then tackle it at a later date

---
title: Dependency Overrides
description: A guide to overriding dependencies set in a mod's `fabric.mod.json`.
authors:
  - cassiancc
  - skycatminepokie
  - ytg1234
authors-nogithub:
  - kb1000
resources:
  https://semver.org/: Semantic Versioning
---

<!---->

::: warning

> Dependency overrides are used to give modpack developers control over their mods. This should not be used by regular players.

Reviewer: Hmmm, why is it in the `players` section then? ig it doesn't really fit under `develop` either

Author: I believe Fabric generally considers modpack developers and normal players to be the same demographic - see `#player-support` serving both rather than having a dedicated `pack-development` channel like NeoForge does.

It is recommended to understand how [mod dependency fields are structured](../../develop/loader/fabric-mod-json#semantic-versioning) before continuing.

:::

When developing a modpack, it is possible to find mods with dependency requirements that do not match what the mod actually requires. A common case of this is mods with too strict of a Minecraft dependency, causing them to unnecessarily break when the game recieves a minor hotfix like 26.1.1.

> To solve this, Fabric Loader allows overriding a mod's dependency requirement so that the mod can attempt to load on a version of Minecraft that it is not designed for. This is meant to be a temporary solution if possible. If the mod is actively maintained, it's recommended to report this incompatibility on its issue tracker so that users do not need to use this workaround.

Reviewer: ```suggestion
Sometimes, when assembling a modpack, you might encounter mods with unhelpful dependency requirements - for example, a mod might be too stringent and require Minecraft `26.1`, despite it also working on `26.1.2`.

To counter this, Fabric Loader lets you override dependency requirements, so you can attempt to load a mod on a version of Minecraft it is not designed for.

::: tip

Overriding dependencies should only be a temporary solution if possible. If the mod is actively maintained, consider reporting this incompatibility on the issue tracker, and letting the developers upstream take care of the issue.

:::
```

## Setup {#setup}

> ::: info

Reviewer: ```suggestion
:::: info
```

this is to clearly show that containers are being indented

For the purposes of this example, we'll be using the following `fabric.mod.json` for a mod with the id `example-mod`. At any point, you can switch tabs in a code block to see how the dependency override affects this `fabric.mod.json`.

::: details `fabric.mod.json`

```json
{
  "depends": {
    "fabricloader": ">=0.11.1",
    "fabric-api": ">=0.28.0",
    "minecraft": "26.1"
  },
  "breaks": {
    "optifabric": "*"
  },
  "suggests": {
    "anothermod": "*",
    "flamingo": "*",
    "modupdater": "*"
  }
}
```

> :::

Reviewer: ```suggestion
:::

::::
```

ditto
